dolphinscheduler-server/src/main/java/org/apache/dolphinscheduler/server/worker/task/datax/DataxTask.java

```diff
@@ -236,64 +222,90 @@ private String buildDataxJsonFile()
 
-        List<JSONObject> readerConnArr = new ArrayList<>();
-        JSONObject readerConn = new JSONObject();
-        readerConn.put("querySql", new String[] {dataXParameters.getSql()});
-        readerConn.put("jdbcUrl", new String[] {dataSourceCfg.getJdbcUrl()});
+        List<ObjectNode> readerConnArr = new ArrayList<>();
+        ObjectNode readerConn = JSONUtils.createObjectNode();
+
+        ArrayNode sqlArr = readerConn.putArray("querySql");
+        for (String sql : new String[]{dataXParameters.getSql()}) {
+            sqlArr.add(sql);
+        }
+
+        ArrayNode urlArr = readerConn.putArray("jdbcUrl");
+        for (String url : new String[]{dataSourceCfg.getJdbcUrl()}) {
+            urlArr.add(url);
+        }
+
         readerConnArr.add(readerConn);
 
-        JSONObject readerParam = new JSONObject();
+        ObjectNode readerParam = JSONUtils.createObjectNode();
         readerParam.put("username", dataSourceCfg.getUser());
         readerParam.put("password", dataSourceCfg.getPassword());
-        readerParam.put("connection", readerConnArr);
+        readerParam.putArray("connection").addAll(readerConnArr);
 
-        JSONObject reader = new JSONObject();
+
+        ObjectNode reader = JSONUtils.createObjectNode();
         reader.put("name", DataxUtils.getReaderPluginName(DbType.of(dataxTaskExecutionContext.getSourcetype())));
-        reader.put("parameter", readerParam);
+        reader.set("parameter", readerParam);
+
+        List<ObjectNode> writerConnArr = new ArrayList<>();
+        ObjectNode writerConn = JSONUtils.createObjectNode();
+        ArrayNode tableArr = writerConn.putArray("table");
+        for (String table : new String[]{dataXParameters.getTargetTable()}) {
+            tableArr.add(table);
+        }
 
-        List<JSONObject> writerConnArr = new ArrayList<>();
-        JSONObject writerConn = new JSONObject();
-        writerConn.put("table", new String[] {dataXParameters.getTargetTable()});
         writerConn.put("jdbcUrl", dataTargetCfg.getJdbcUrl());
         writerConnArr.add(writerConn);
 
-        JSONObject writerParam = new JSONObject();
+        ObjectNode writerParam = JSONUtils.createObjectNode();
         writerParam.put("username", dataTargetCfg.getUser());
         writerParam.put("password", dataTargetCfg.getPassword());
-        writerParam.put("column",
-            parsingSqlColumnNames(DbType.of(dataxTaskExecutionContext.getSourcetype()),
-                    DbType.of(dataxTaskExecutionContext.getTargetType()),
-                    dataSourceCfg, dataXParameters.getSql()));
-        writerParam.put("connection", writerConnArr);
+
+        String[] columns = parsingSqlColumnNames(DbType.of(dataxTaskExecutionContext.getSourcetype()),
+                DbType.of(dataxTaskExecutionContext.getTargetType()),
+                dataSourceCfg, dataXParameters.getSql());
+        ArrayNode columnArr = writerParam.putArray("column");
```

Comment:
why not use addAll

Reply:
The type of columns is String[] , addAll() only accept ArrayNode